ecocheck.py

```diff
@@ -0,0 +1,118 @@
+import logging
+from infra.system.helpers.utils import getScriptDir
+from infra.system.helpers.schema_validator import SchemaValidator
+import time
+import datetime
+from gofed.lib.pkgdb2client import get_packages
+from gofed.lib.bitbucketclient import get_commits_bitbucket
+from gofed.lib.githubclient import get_commits_github
+from gofed.lib.kojiclient import KojiClient
+
+
+SUPPORTED_REPOSIRORIES = ['github.com', 'bitbucket.org']
+
+
+def repository_info(clone_url):
+	if not clone_url:
+		return []
+	data = clone_url.split('/')
+	if data[2] in SUPPORTED_REPOSIRORIES:
+		repository = {
+			'host': data[2],
+			'owner': data[3],
+			'package': data[4].split('.')[0]
+		}
+	else:
+		logging.error('Unsupported upstream host: %s' % data[2])
+		return []
+	return repository
+
+
+class EcoCheck:
```

Comment:
There are two things to consider:

* EcoCheck as a component
* EcoCheck as an act

In the former case, the aim is to create a component that will give me a list of what is new in an ecosystem. The product/output is a dictionary I will forward to the EcoCheck act.

In the latter case, the aim is to create an act. The act is responsible for carrying all steps needed to extract data from a list of what is new.

Here is mine suggestion:
- create EcoCheck class that will combine all clients. It will return the dictionary as you defined by the JSON Schema. The class will have scan(last_change_date) method (or check) what will return the dictionary.
- create EcoCheckAct that will take EcoCheck component (and in future it will take more components, each responsible for an area of an ecosystem). The act will then carry its operation to extract data.

EcoCheck class can then go under gofed/lib and can be used by other projects or components. The JSON Schema is still valid here. At the same time EcoCheck can provide three methods (as I mention below): ``getNewPackages``, ``getNewCommits`` and ``getNewBuilds`` which can be used independent of each other and by other components/classes.